keyring/keychain.go

```diff
@@ -37,483 +20,131 @@ func init() {
 			name = "login"
 		}
 
-		home, err := homedir.Dir()
-		if err != nil {
-			return nil, err
-		}
-
 		return &keychain{
-			path:    filepath.Join(home, "/Library/Keychains/"+name+".keychain"),
```

Comment:
This is one change that probably needs testing. The new code uses simply name +".keychain", vs the old one that references the full path. It works fine on my machine which is 10.12.6.